### PR TITLE
Fix/7625 product share info

### PR DESCRIPTION
**Pull request by richbrennan**
Closes https://github.com/mozilla/foundation.mozilla.org/issues/7625

**Link to sample test page**: https://foundation-s-fix-7625-p-jsfyhx.mofostaging.net/en/privacynotincluded/leg-quality-pay-keep/

### CODE REVIEW

File: network-api/networkapi/wagtailpages/templates/buyersguide/product_page.html [hunk 1]
```diff
@@ -2,21 +2,10 @@
 
 {% load wagtailcore_tags bg_selector_tags env l10n i18n localization static wagtailimages_tags %}
 
-{% block head_extra %}
```
Comment (richbrennan) [inline network-api/networkapi/wagtailpages/templates/buyersguide/product_page.html:5]: This block doesn't seem to exist anywhere else in the codebase.

File: network-api/networkapi/wagtailpages/templates/buyersguide/product_page.html [hunk 2]
```diff
@@ -2,21 +2,10 @@
...
-  <meta property="og:title" content="{% blocktrans context "This can be localized. This is a reference to the “*batteries not included” mention on toys." %}privacy not included - {{ product.title }}{% endblocktrans %}" />
-  <link rel="canonical" href="{{ request.scheme }}://{{ request.get_host }}{{ request.get_full_path }}" />
-{% endblock %}
-
-{% block social_image %}
```
Comment (richbrennan) [inline network-api/networkapi/wagtailpages/templates/buyersguide/product_page.html:10]: This block doesn't seem to exist anywhere else in the codebase.

File: network-api/networkapi/wagtailpages/templates/buyersguide/product_page.html [hunk 3]
```diff
@@ -2,21 +2,10 @@
...
-  {% if page.search_image %}
-    {% image page.search_image original as search_image %}
-    <meta property="og:image" content="{{search_image.url}}">
-    <meta name="twitter:image" content="{{search_image.url}}">
-  {% else %}
-    {% static "_images/buyers-guide/evergreen-social.png" as default_image_url %}
-    <meta property="og:image" content="{{default_image_url}}">
-    <meta name="twitter:image" content="{{default_image_url}}">
-  {% endif %}
+{% block social_metadata %}
```
Comment (richbrennan) [inline network-api/networkapi/wagtailpages/templates/buyersguide/product_page.html:5]: Overridden to remove extra `og:title`

File: network-api/networkapi/wagtailpages/pagemodels/products.py [hunk 4]
```diff
@@ -743,13 +745,36 @@ def get_voting_json(self):
...
+    def get_meta_description(self):
+        if self.search_description:
+            return self.search_description
+
+        soup = BeautifulSoup(self.blurb, "html.parser")
+        first_paragraph = soup.find("p")
+        if first_paragraph:
+            return first_paragraph.text
+
+        return super().get_meta_description()
```
Comment (richbrennan) [inline network-api/networkapi/wagtailpages/pagemodels/products.py:768]: The parent classes method also runs
```if self.search_description:
        return self.search_description
```
But the small repetition seemed better than copying the code

**Comment by danielfmiranda** [comment]
Hi everyone, just wanted to mention that this seems to be working correctly:

Here is what the PNI product page would look like on twitter:
image

and in slack:
image

**Comment by kristinashu** [comment]
Thank you for the screen caps Daniel! This is looking good - seems like it's pulling the title, intro text, and image correctly.

Anyone know why Slack is showing "Foundation Home Page"?

Blog pages in Slack say "Mozilla Foundation" which seems better than "Foundation Home Page".

image

**Review by kristinashu — APPROVED** [review]
Awesome, thank you Rich!

**Review by TheoChevalier — CHANGES_REQUESTED** [review]

**Review by TheoChevalier — APPROVED** [review]
L10n r+!

**Review by Pomax — CHANGES_REQUESTED** [review]
Is there a way we can use https://github.com/mozilla/foundation.mozilla.org/blob/main/network-api/networkapi/wagtailpages/pagemodels/mixin/foundation_metadata.py here?

**Comment by richbrennan** [comment]
> Is there a way we can use https://github.com/mozilla/foundation.mozilla.org/blob/main/network-api/networkapi/wagtailpages/pagemodels/mixin/foundation_metadata.py here instead?

I'm not sure I follow, as I kind of am using the mixin by adding/overriding the methods defined by the `wagtail-metadata` package as required for the page. I'd be reluctant to try and shoehorn my page-specific logic into the `FoundationMetadataPageMixin` class, keeping it on the page itself seems most appropriate given it only applies to product pages, or have I missed something?

**Comment by Pomax** [comment]
I've filed https://github.com/mozilla/foundation.mozilla.org/issues/7828 to look at whether we can consolidate and delegate metadata, so we can look at that separately from this specific PR

**Review by Pomax — CHANGES_REQUESTED** [review]
R+ given https://github.com/mozilla/foundation.mozilla.org/issues/7828, but it looks like there's a merge conflict that needs fixing before we can land this.

**Comment by richbrennan** [comment]
Gah, broken by my own other ticket. Have fixed and pushed

**Review by Pomax — APPROVED** [review]

**Pull request merged**